### PR TITLE
Add an edit button to the profile links section

### PR DESCRIPTION
**Pull request by iptq**
Trying out something simple to learn the codebase, so I'd appreciate any kind of advice on this.

This patch adds an Edit button to the right side of the links section, as requested in https://github.com/ppy/osu-web/issues/2455. When clicked, it just navigates to the account edit page where the user can make the changes they wanted to make.

image

This is what it looks like when there's only one row:

image

Closes https://github.com/ppy/osu-web/issues/2455

### CODE REVIEW

File: resources/assets/less/bem/profile-links.less [hunk 1]
```diff
@@ -36,4 +42,27 @@
     font-weight: 600;
     word-wrap: break-word;
   }
+
+  .edit-profile-links {
```
Comment (nanaya) [inline resources/assets/less/bem/profile-links.less:46]: this should conform with [bem](https://css-tricks.com/bem-101/). Either as a new block (in its own file) or as element of `profile-links` (`profile-links__actions`)

File: resources/assets/less/bem/profile-links.less [hunk 2]
```diff
@@ -36,4 +42,27 @@
...
+    position: absolute;
+    top: 0;
+    right: 0;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 70px;
```
Comment (nanaya) [inline resources/assets/less/bem/profile-links.less:53]: this should be `@gutter-profile-page` on default and `@gutter-profile-page-desktop` on desktop. Check `page-extra__actions`.

File: resources/assets/coffee/react/profile-page/links.coffee [hunk 3]
```diff
@@ -103,6 +103,14 @@ export class Links extends React.PureComponent
       for row, j in rows when _.compact(row).length > 0
         div key: j, className: "#{bn}__row #{bn}__row--#{j}", row
 
+      div
+        className: 'edit-profile-links'
+        a
+          href: laroute.route('account.edit')
+          button
```
Comment (nanaya) [inline resources/assets/coffee/react/profile-page/links.coffee:110]: `button` can't be inside `a`. I don't see the point anyway (apart of skipping some styling).

File: resources/assets/less/bem/profile-links.less [hunk 4]
```diff
@@ -36,4 +42,27 @@
...
+      color: @osu-colour-c1;
+      button {
+        background-color: @osu-colour-b2;
+        outline: none;
+        border: none;
+        border-radius: 50%;
+        width: 40px;
+        height: 40px;
+      }
+    }
```
Comment (nanaya) [inline resources/assets/less/bem/profile-links.less:66]: just use `profile-page-toggle`. The styling will need a bit of update so it also works with `a` element.

**Review by nanaya — DISMISSED** [review]

**Comment by peppy** [comment]
@arflyte can you confirm you're okay with there being three permanently displaying edit buttons on the profile page with this addition?

**Comment by iptq** [comment]
Oh shoot, I meant for the edit button to only display in case the user
is viewing their own userpage. I'll implement that tomorrow.

**Comment by peppy** [comment]
I am presuming that is already the case (haven't checked your logic). Even in such a case, there are now three edit buttons in total (see two existing ones in screenshot):

20210208 190205 (Safari app)

**Comment by arflyte** [comment]
Since it's just for the player's own profile, yeah. Although I definitely need to refine the edit button a lil bit more.

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**